src/analysis/normed_space/dual.lean

```diff
@@ -96,3 +109,80 @@ end
 end bidual_isometry
 
 end normed_space
+
+namespace inner_product_space
+open is_R_or_C continuous_linear_map
+
+variables (𝕜 : Type*)
+variables {E : Type*} [is_R_or_C 𝕜] [inner_product_space 𝕜 E]
+local notation `⟪`x`, `y`⟫` := @inner 𝕜 E _ x y
+local postfix `†`:90 := @is_R_or_C.conj 𝕜 _
+
+/--
+Given some x in an inner product space, we can define its dual as the continuous linear map
+λ y, ⟪x, y⟫.
+-/
+def to_dual (x : E) : normed_space.dual 𝕜 E :=
+linear_map.mk_continuous
+{ to_fun := λ y, ⟪x, y⟫,
+  map_add' := by simp only [inner_add_right, forall_const, eq_self_iff_true],
+  map_smul' := by simp [inner_smul_right] }
+∥x∥
+(λ y, by { rw [is_R_or_C.norm_eq_abs], exact abs_inner_le_norm _ _ })
+
+variables {𝕜}
+
+@[simp] lemma to_dual_zero : to_dual 𝕜 (0 : E) = 0 :=
+by { ext, simp [to_dual] }
+
+/--
+Fréchet-Riesz representation: any ℓ in the dual of a Hilbert space E is of the form
+λ u, ⟪y, u⟫ for some y in E.
+-/
+lemma exists_elem_of_mem_dual [complete_space E] (ℓ : normed_space.dual 𝕜 E) :
+  ∃ y : E, ℓ = to_dual 𝕜 y :=
+begin
+  set Y := ker ℓ with hY,
+  by_cases htriv : Y = ⊤,
+  { have hℓ : ℓ = 0,
+    { have h' := linear_map.ker_eq_top.mp htriv,
+      rw [←coe_zero] at h',
+      apply coe_injective,
+      exact h' },
+    exact ⟨0, by simp [hℓ]⟩ },
+  { have Ycomplete := is_complete_ker ℓ,
+    rw [submodule.eq_top_iff_orthogonal_eq_bot Ycomplete, ←hY] at htriv,
+    change Y.orthogonal ≠ ⊥ at htriv,
+    rw [submodule.ne_bot_iff] at htriv,
+    obtain ⟨z : E, hz : z ∈ Y.orthogonal, z_ne_0 : z ≠ 0⟩ := htriv,
+    refine ⟨((ℓ z)† / ⟪z, z⟫) • z, _⟩,
+    ext x,
+    have h₁ : (ℓ z) • x - (ℓ x) • z ∈ Y,
+    { rw [mem_ker, map_sub, map_smul, map_smul, algebra.id.smul_eq_mul, algebra.id.smul_eq_mul,
+          mul_comm],
+      exact sub_self (ℓ x * ℓ z) },
+    have h₂ : (ℓ z) * ⟪z, x⟫ = (ℓ x) * ⟪z, z⟫,
+    { have h₃ := calc
+        0    = ⟪z, (ℓ z) • x - (ℓ x) • z⟫       : by { rw [(Y.mem_orthogonal' z).mp hz], exact h₁ }
+         ... = ⟪z, (ℓ z) • x⟫ - ⟪z, (ℓ x) • z⟫  : by rw [inner_sub_right]
+         ... = (ℓ z) * ⟪z, x⟫ - (ℓ x) * ⟪z, z⟫  : by simp [inner_smul_right],
+      exact sub_eq_zero.mp (eq.symm h₃) },
+    dsimp [to_dual],
```

Comment:
Add a simp lemma that will perform this transformation for you?

Reply:
Done! (Actually I think that line was not even needed in the end.)